src/monitor.cpp

```diff
@@ -225,29 +225,31 @@ double structure::get_chi1inv(component c, direction d, const ivec &origloc, dou
   return 0.0;
 }
 
-/* Set Vinv = inverse of V, where both V and Vinv are real-symmetric matrices.*/
-void matrix_invert(std::vector<double> &Vinv, std::vector<double> &V) {
+/* Set Vinv = inverse of V, where both V and Vinv are complex matrices.*/
+void matrix_invert(std::complex<double> (&Vinv)[9], std::complex<double> (&V)[9]) {
   
-  double det = (V[0 +3*0] * (V[1 + 3*1]*V[2 +3*2] - V[1 + 3*2]*V[2 +3*1]) - 
+  std::complex<double> det = (V[0 +3*0] * (V[1 + 3*1]*V[2 +3*2] - V[1 + 3*2]*V[2 +3*1]) - 
   V[0 + 3*1] * (V[0 + 3*1]*V[2 + 3*2] - V[1 + 3*2]*V[0 + 3*2]) + 
   V[0 + 3*2] * (V[0 + 3*1]*V[1 + 3*2] - V[1 + 3*1]*V[0 + 3*2]));
 
-  if (det == 0) abort("meep: Matrix is singular, aborting.\n");
-
-  Vinv[0 + 3*0] = 1/det * (V[1 + 3*1]*V[2 + 3*2] - V[1 + 3*2]*V[2 + 3*1]);
-  Vinv[0 + 3*1] = 1/det * (V[0 + 3*2]*V[2 + 3*1] - V[0 + 3*1]*V[2 + 3*2]);
-  Vinv[0 + 3*2] = 1/det * (V[0 + 3*1]*V[1 + 3*2] - V[0 + 3*2]*V[1 + 3*1]);
-  Vinv[1 + 3*0] = 1/det * (V[1 + 3*2]*V[2 + 3*0] - V[1 + 3*0]*V[2 + 3*2]);
-  Vinv[1 + 3*1] = 1/det * (V[0 + 3*0]*V[2 + 3*2] - V[0 + 3*2]*V[2 + 3*0]);
-  Vinv[1 + 3*2] = 1/det * (V[0 + 3*2]*V[1 + 3*0] - V[0 + 3*0]*V[1 + 3*2]);
-  Vinv[2 + 3*0] = 1/det * (V[1 + 3*0]*V[2 + 3*1] - V[1 + 3*1]*V[2 + 3*0]);
-  Vinv[2 + 3*1] = 1/det * (V[0 + 3*1]*V[2 + 3*0] - V[0 + 3*0]*V[2 + 3*1]);
-  Vinv[2 + 3*2] = 1/det * (V[0 + 3*0]*V[1 + 3*1] - V[0 + 3*1]*V[1 + 3*0]);
+  if (det.real() == 0 && det.imag() == 0) abort("meep: Matrix is singular, aborting.\n");
```

Comment:
doesn't `det == 0.0` work?

Reply:
I was just being safe -- do you want it changed?

Reply:
`det == 0.0` seems clearer and is safe.

(You can't do `det == 0`, on the other hand, because C++ doesn't let you mix `complex<double>` with scalars of other types.)

Reply:
Just fixed with 8590db1